RecoMET/METFilters/plugins/EcalBadCalibFilter.cc

```diff
@@ -0,0 +1,171 @@
+#include "DataFormats/DetId/interface/DetId.h"
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+ 
+#include "Geometry/CaloTopology/interface/CaloTopology.h"
+#include "Geometry/CaloEventSetup/interface/CaloTopologyRecord.h"
+ 
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+ 
+using namespace std;
+ 
+ 
+class ecalBadCalibSep2017ListFilter : public edm::global::EDFilter<> {
+ 
+public:
+ 
+  explicit ecalBadCalibSep2017ListFilter(const edm::ParameterSet & iConfig);
+  ~ecalBadCalibSep2017ListFilter() override {}
+ 
+private:
+ 
+  // main filter function
+ 
+ bool filter(edm::StreamID, edm::Event & iEvent, const edm::EventSetup & iSetup) const override; 
+ 
+  // input parameters
+ 
+  // eb rechit collection (from AOD)
+  const edm::EDGetTokenT<EcalRecHitCollection>  ebRHSrcToken_;  
+  // ee rechit collection (from AOD)
+  const edm::EDGetTokenT<EcalRecHitCollection>  eeRHSrcToken_;
+
+  //config parameters (defining the cuts on the bad SCs)
+  const double ebMin_;              // EB rechit et threshold
+  const double eeMin_;              // EE rechit et threshold
+ 
+  const std::vector<unsigned int> baddetEB_;    // DetIds of bad EB channels  
+  const std::vector<unsigned int> baddetEE_;    // DetIds of bad EE channels
+ 
+  const bool taggingMode_;
+  const bool debug_;                // prints out debug info if set to true
+ 
+};
+ 
+// read the parameters from the config file
+ecalBadCalibSep2017ListFilter::ecalBadCalibSep2017ListFilter(const edm::ParameterSet & iConfig)
+  : ebRHSrcToken_     (consumes<EcalRecHitCollection>(iConfig.getParameter<edm::InputTag>("EBRecHitSource")))
+  , eeRHSrcToken_     (consumes<EcalRecHitCollection>(iConfig.getParameter<edm::InputTag>("EERecHitSource")))
+  , ebMin_          (iConfig.getParameter<double>("ebMinEt"))
+  , eeMin_          (iConfig.getParameter<double>("eeMinEt"))
+  , baddetEB_       (iConfig.getParameter<std::vector<unsigned int> >("baddetEB"))
+  , baddetEE_       (iConfig.getParameter<std::vector<unsigned int> >("baddetEE"))
+  , taggingMode_    (iConfig.getParameter<bool>("taggingMode"))
+  , debug_          (iConfig.getParameter<bool>("debug"))
+{
+  produces<bool>();
+}
+ 
+
+bool ecalBadCalibSep2017ListFilter::filter(edm::StreamID, edm::Event & iEvent, const edm::EventSetup & iSetup) const {
+
+
+  // load required collections
+ 
+  // EB rechit collection
+  edm::Handle<EcalRecHitCollection> ebRHs;
+  iEvent.getByToken(eeRHSrcToken_, ebRHs);
+ 
+  // EE rechit collection
+  edm::Handle<EcalRecHitCollection> eeRHs;
+  iEvent.getByToken(eeRHSrcToken_, eeRHs);
+ 
+  // Calo Geometry - needed for computing E_t
+  edm::ESHandle<CaloGeometry> pG;
+  iSetup.get<CaloGeometryRecord>().get(pG);
+ 
+  
+  // by default the event is OK
+  bool pass = true;
+ 
+ 
+ 
+  // define energy variables and ix,iy,iz coordinates
+  int ix,iy,iz;
+  ix=0,iy=0,iz=0;
+  float ene=0;
+  float et=0;
+ 
+  for (const auto eeit : baddetEE_) {
```

Comment:
Where is the loop for EB?

Reply:
@slava77 Issues with the calibration were observed only in EE [for now at least]

Reply:
OK, it sounds like there should only be one baddet vector and only one configured input.
In the specific configuration you can then specify EE (the instance should have EE in the name in this case) and if EB filtering is needed in the future we'll just need to change the configuration filter.
This way we have less code replication (no second loop over EB in this file) and only configuration change is needed to update.

Reply:
@slava77 Makes sense - thank you. I will update the code accordingly

Reply:
the c++ code still uses EE.
This will become rather confusing if we decide to use the same C++ code configured in python to use EB.

Reply:
@slava77 I fixed the names but need to fix also this : EEDetId ecaldet(ecalit); to be pretty transparent - will update the PR later today

Reply:
ah, good point.
hit collection can still be searched by a plain DetId, right?

Reply:
replying to self: yes
The EcalRecHit collection keys are defined by DetId
https://github.com/cms-sw/cmssw/blob/master/DataFormats/EcalRecHit/interface/EcalRecHit.h#L17